openmdao/core/group.py

```diff
@@ -56,16 +56,51 @@ def __init__(self, **kwargs):
 
     def setup(self):
         """
-        Add subsystems to this group.
+        Build this group.
+
+        This method should be overidden by your Group's method.
+
+        You may call 'add_subsystem' to add systems to this group. You may also issue connections,
+        and set the linear and nonlinear solvers for this group level. You cannot safely change
+        anything on children systems; use the 'configure' method instead.
+
+        Available attributes:
+            name
+            pathname
+            comm
+            metadata
+        """
+        pass
+
+    def configure(self):
+        """
+        Configure this group to assign children settings.
+
+        This method may optionally be overidden by your Group's method.
+
+        You may only use this method to change settings on your children subsystems. This includes
+        setting solvers in cases where you want to override the defaults.
```

Comment:
I think we should include some detail about timing here. 

"You can assume that the full hierarchy below your level has been instantiated and has already called its own configure methods."